### PR TITLE
flat group creation

### PR DESCRIPTION
**Pull request by dnbrwstr**
This pr simplifies group creation and allows us to use generated titles for untitled groups. Touches quite a lot, so any qa-ing would be appreciated.

Fixes TLON-2794

### CODE REVIEW

File: packages/ui/src/hooks/groupsSorters.ts [hunk 1]
```diff
@@ -50,60 +51,29 @@ export function useAlphabeticallySegmentedGroups({
...
-
-  try {
-    const aTitleClean = a.id.split('/')[1].replace(/[~-]/g, '');
-    const bTitleClean = b.id.split('/')[1].replace(/[~-]/g, '');
-    return aTitleClean.localeCompare(bTitleClean);
-  } catch (e) {
-    return 0;
-  }
+function groupsSorter(a: GroupSegment, b: GroupSegment): number {
+  return a.sortable.localeCompare(b.sortable);
```
Comment (latter-bolden) [inline packages/ui/src/hooks/groupsSorters.ts:78]: doesn't seem like this would do what's expected?

File: packages/ui/src/utils/channelUtils.tsx [hunk 2]
```diff
@@ -13,15 +14,15 @@ export function getChannelMemberName(
   if (disableNicknames) {
     return member.contactId;
   }
-  return member.contact?.nickname ?? member.contactId;
+  return member.contact?.nickname ?? formatUserId(member.contactId)?.display;
```
Comment (latter-bolden) [inline packages/ui/src/utils/channelUtils.tsx:17]: we've run into null coalesce issues in these kinds of comparisons if nickname is set but it's an empty string — can we make sure we defend against that?

File: packages/ui/src/components/Channel/EmptyChannelNotice.tsx [hunk 3]
```diff
@@ -13,33 +17,42 @@ export function EmptyChannelNotice({
   channel: db.Channel;
   userId: string;
 }) {
-  const isGroupAdmin = useIsAdmin(channel.groupId ?? '', userId);
+  const { onPressGroupMeta } = useChatOptions() ?? {};
```
Comment (latter-bolden) [inline packages/ui/src/components/Channel/EmptyChannelNotice.tsx:20]: When do you expect `useChatOptions` to not return anything? Only when the provider is uninitialized?

Trying locally on a freshly created multi-dm, this seems to no-op

**Review by latter-bolden — COMMENTED** [review]
Let's chat through some of the backcompat implications tomorrow

File: packages/ui/src/components/Avatar.tsx [hunk 4]
```diff
@@ -110,9 +110,15 @@ export const GroupAvatar = React.memo(function GroupAvatarComponent({
...
+  const { disableNicknames } = useCalm();
+  const fallbackTitle = useMemo(() => {
+    return isGroupImageShim(model)
+      ? model.title
+      : utils.getGroupTitle(model, disableNicknames);
+  }, [disableNicknames, model]);
   const fallback = (
     <TextAvatar
-      text={model.title ?? model.id.replace('~', '')}
+      text={fallbackTitle ?? 'G'}
```
[the remark this answers is not in the document]
Comment (dnbrwstr) [reply packages/ui/src/components/Avatar.tsx:121]: Lmao, got another suggestion?

File: packages/ui/src/components/Channel/EmptyChannelNotice.tsx (hunk 3 above)
Comment (dnbrwstr) [reply packages/ui/src/components/Channel/EmptyChannelNotice.tsx:20]: Ya, never, the hook should actually throw an error if that's the case -- good catch.

File: packages/ui/src/components/Avatar.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jamesacklin) [reply packages/ui/src/components/Avatar.tsx:121]: ~

[the remark this answers is not in the document]
Comment (latter-bolden) [reply packages/ui/src/components/Avatar.tsx:121]: No that's fine lol it just made me giggle

**Comment by jamesacklin** [comment]
A few things of note, ordered in reverse-severity (most urgent stuff first).

The "Customize" button on the initial channel creation view is inactive for quite some time, until I open the "..." menu and navigate back. Furthermore, long-pressing the group in the ChatList and selecting "Edit Group" does nothing.

https://github.com/user-attachments/assets/82a1ab03-810c-4cd9-87bb-b59d7b3d8f50

---

Both variants of the second step in the creation flow need some extra copy. Why am I seeing a list of people here? For the DM case, we need to add rightward-carets to show that choosing an option will navigate the user.

Screenshot 2024-12-11 at 3 34 05 PM

Screenshot 2024-12-11 at 3 36 32 PM

---

We are missing safe-area padding on the "Edit group metadata" screen and the inviting-members screen in the creation flow.

Screenshot 2024-12-11 at 3 35 02 PM

Screenshot 2024-12-11 at 3 34 05 PM

---

In the GroupOptions sheet only accessible from inside the channel, the "Mark as Read" option appears even though there are no messages in the group.

Screenshot 2024-12-11 at 3 36 07 PM


---

The initial bifurcated creation sheet has some text truncation issues in the description of the second item. I'd recommend a copy change.

Screenshot 2024-12-11 at 3 33 51 PM

**Comment by dnbrwstr** [comment]
Re-requesting review; I've added a placeholder title for backcompat + worked through most of the ux changes.

The only thing I didn't add was right arrows for contacts in the new dm sheet -- looking around at other apps, that's usually used to indicate that the row will navigate within the sheet, not when selecting the row is a primary action.

**Comment by jamesacklin** [comment]
> looking around at other apps, that's usually used to indicate that the row will navigate within the sheet, not when selecting the row is a primary action.

Rationale checks out, thank you chef

**Review by latter-bolden — APPROVED** [review]
Changes lgtm, let's get it in!

**Pull request merged**